core/src/test/java/apoc/trigger/TriggerNewProceduresTest.java

```diff
@@ -143,6 +144,28 @@ public void testRemoveNode() {
                 db, "MATCH (c:Counter) RETURN c.count as count", (row) -> assertEquals(1L, row.get("count")), TIMEOUT);
     }
 
+    @Test
+    public void testRemoveNode2() {
+        db.executeTransactionally("CREATE (f:Foo {prop: 1})");
+
+        String name = "access-removed-node";
+        String query = "CREATE (:TEST {data: apoc.convert.toJson($deletedNodes)})";
+        sysDb.executeTransactionally(
+                "CALL apoc.trigger.install('neo4j', $name, $query, {phase: 'afterAsync'})",
+                Map.of("name", name, "query", query));
+        awaitTriggerDiscovered(db, name, query);
+
+        db.executeTransactionally("MATCH (f:Foo) DETACH DELETE f");
+        Util.sleep(500);
```

Comment:
I would prefer to use the await until asserted thing, but we can always switch if it becomes flaky.

Reply:
Ah yes, I had made a mental note to do that, and forgot, so thanks! will fix up